### PR TITLE
ci: Added infraction tests to provider suite

### PR DESCRIPTION
**Pull request by stana-miric**
This pr includessome interchain test changes:
- Added tests for customizable infraction parameters that were implemented [here](https://github.com/cosmos/interchain-security/pull/2403). 
- Split provider suites with single and multiple validators
- Enables running tests with custom image name and tag so that we can also run test on e.g. local built image. Added flags are PROVIDER_IMAGE_NAME and PROVIDER_IMAGE_TAG. Default values if we omit the flags are "ghcr.io/cosmos/interchain-security" and "latest". 
E.g. running cmd: make test-interchain PROVIDER_IMAGE_NAME=ics-local PROVIDER_IMAGE_TAG=v1

### CODE REVIEW

File: tests/interchain/provider_single_val_test.go [hunk 1]
```diff
@@ -658,3 +618,127 @@ func (s *ProviderSuite) TestProviderOwnerChecks() {
...
+		InfractionParameters: infractionParamsTemplate(),
+	}
+	s.Require().NoError(s.Provider.UpdateConsumer(s.GetContext(), upgradeMsg, testAccKey))
+	s.Require().NoError(testutil.WaitForBlocks(s.GetContext(), 1, s.Provider))
+	consumerChain, err = s.Provider.GetConsumerChain(s.GetContext(), consumerId)
+	s.Require().NoError(err)
+	// chain is in pre-launched phase, params are updated immediatelly
+	s.Require().Equal(infractionParamsTemplate(), convertJsonToInfractionParameters(consumerChain.InfractionParams))
+
+	// Confirm that a chain can be created with custom infrection parameters set only for double sign
```
Comment (mpoke) [inline tests/interchain/provider_single_val_test.go:652]:
```suggestion
	// Confirm that a chain can be created with custom infraction parameters set only for double sign
```

File: tests/interchain/provider_single_val_test.go [hunk 2]
```diff
@@ -658,3 +618,127 @@ func (s *ProviderSuite) TestProviderOwnerChecks() {
...
+	createConsumerMsg = msgCreateConsumer(chainName, nil, nil, infractionParams, testAcc)
+	consumerId, err = s.Provider.CreateConsumer(s.GetContext(), createConsumerMsg, testAccKey)
+	s.Require().NoError(err)
+	consumerChain, err = s.Provider.GetConsumerChain(s.GetContext(), consumerId)
+	s.Require().NoError(err)
+	response := convertJsonToInfractionParameters(consumerChain.InfractionParams)
+	s.Require().Equal(infractionParams.DoubleSign, response.DoubleSign)
+	s.Require().Equal(defaultInfractionParams.Downtime, response.Downtime)
+
+	// Confirm that a chain can be created with custom infrection parameters set only for downtime
```
Comment (mpoke) [inline tests/interchain/provider_single_val_test.go:665]:
```suggestion
	// Confirm that a chain can be created with custom infraction parameters set only for downtime
```

File: tests/interchain/provider_single_val_test.go [hunk 3]
```diff
@@ -658,3 +618,127 @@ func (s *ProviderSuite) TestProviderOwnerChecks() {
...
+	createConsumerMsg = msgCreateConsumer(chainName, nil, nil, infractionParams, testAcc)
+	consumerId, err = s.Provider.CreateConsumer(s.GetContext(), createConsumerMsg, testAccKey)
+	s.Require().NoError(err)
+	consumerChain, err = s.Provider.GetConsumerChain(s.GetContext(), consumerId)
+	s.Require().NoError(err)
+	response = convertJsonToInfractionParameters(consumerChain.InfractionParams)
+	s.Require().Equal(infractionParams.Downtime, response.Downtime)
+	s.Require().Equal(defaultInfractionParams.DoubleSign, response.DoubleSign)
+
+	// Confirm that a chain can be created with custom infrection parameters set for both downtime and doublesign
```
Comment (mpoke) [inline tests/interchain/provider_single_val_test.go:678]:
```suggestion
	// Confirm that a chain can be created with custom infraction parameters set for both downtime and doublesign
```

File: tests/interchain/provider_single_val_test.go [hunk 4]
```diff
@@ -658,3 +618,127 @@ func (s *ProviderSuite) TestProviderOwnerChecks() {
 	s.Require().Equal(powerShapingParams.Top_N, uint32(consumerChain.PowerShapingParams.TopN))
 	s.Require().Equal(chainsuite.GovModuleAddress, consumerChain.OwnerAddress)
 }
+
+// Tests adding  and updating infraction parameters with MsgCreateConsumer, MsgUpdateConsumer
```
Comment (mpoke) [inline tests/interchain/provider_single_val_test.go:622]: Add the scenarios tested in the docstring.

**Review by mpoke — APPROVED** [review]
LGTM

**Review by MSalopek — APPROVED** [review]

File: tests/interchain/provider_single_val_test.go (hunk 4 above)
Comment (stana-miric) [reply tests/interchain/provider_single_val_test.go]: fixed [3a2f0e2](https://github.com/cosmos/interchain-security/pull/2416/commits/3a2f0e2864a2eaac768b0d8a63f45cb817e23b79)

**Pull request merged**